docs/src/ideal.md

```diff
@@ -0,0 +1,180 @@
+```@meta
+CurrentModule = AbstractAlgebra
+DocTestSetup = quote
+    using AbstractAlgebra
+end
+```
+
+# Ideal functionality
+
+AbstractAlgebra.jl provides a module, implemented in `src/generic/Ideal.jl` for
+ideals of a Euclidean domain or of a univariate or multivariate polynomial ring
+over a Euclidean domain, assuming the existence of a `gcdx` function.
+
+## Generic ideal types
+
+AbstractAlgebra.jl provides a generic ideal type based on Julia arrays which
+is implemented in `src/generic/Ideal.jl`.
+
+These generic ideals have type `Generic.Ideal{T}` where `T` is the type of
+elements of the ring the ideals belong to. Internally they consist of a Julia
+array of generators and some additional fields for a parent object, etc. See
+the file `src/generic/GenericTypes.jl` for details.
+
+Parent objects of ideals have type `Generic.IdealSet{T}`.
```

Comment:
So, just to understand: Is the parent of an ideal meant to stand for "the set of all ideals over rings with matching type"?

From a technical point of view: what are parents needed for / what could they be used for?

I note that ideals in Oscar don't seem to have parents (and that never bothered me so far, but then again maybe it's because I never had them, and so I never thought about it).

Reply:
Yes. We also have them for other ideal types in Hecke. They are useful/necessary when defining maps/morphisms where the domain/codomain consists of ideals.

Reply:
Aha, @fieker just told me that this can be useful to define homomorphisms from sets of ideals to other such sets. OK, makes sense, I guess